src/nbl/asset/utils/IGLSLCompiler.cpp

```diff
@@ -180,16 +180,16 @@ namespace impl
                 //While including a filesystem file it must be either absolute path (or relative to any search dir added to asset::iIncludeHandler; <>-type),
                 //  or path relative to executable's working directory (""-type).
                 relDir = io::IFileSystem::getFileDir(_requesting_source);
-                if (relDir.lastChar() != '/')
-                    relDir.append('/');
+                if (*relDir.string().rbegin() != '/')
+                    relDir += '/';
             }
 
-            io::path name = (_type == shaderc_include_type_relative) ? (relDir + _requested_source) : (_requested_source);
+            std::filesystem::path name = (_type == shaderc_include_type_relative) ? (relDir.string() + _requested_source) : (_requested_source);
             if (!reqBuiltin)
                 name = m_fs->getAbsolutePath(name);
 
             if (_type == shaderc_include_type_relative)
-                res_str = m_inclHandler->getIncludeRelative(_requested_source, relDir.c_str());
+                res_str = m_inclHandler->getIncludeRelative(_requested_source, relDir.string());
```

Comment:
`getIncludeRelative` signature should be changed to take a `std::filesystem:path` instead of a string